docs/file-reference/index.md

```diff
@@ -0,0 +1,628 @@
+---
+title: Odo Devfile Reference
+
+language_tabs:
+  - yaml
+
+toc_footers:
+  - <a href='https://odo.dev'>Odo Homepage</a>
+  - <a href='https://devfile.github.io/devfile/index.html'>Devfile GitHub Page</a>
+
+search: true
+---
+
+# Introduction
+
+> A full odo Devfile example using Spring Boot
+
+```yaml
+ 
+schemaVersion: 2.0.0
+metadata:
+  name: springBoot
+  version: 1.0.0
+projects:
+  - name: springbootproject
+    git:
+      location: "https://github.com/odo-devfiles/springboot-ex.git"
+components:
+  - container:
+      name: tools
+      image: maysunfaisal/springbootbuild
+      memoryLimit: 768Mi
+      command: ['tail']
+      args: [ '-f', '/dev/null']
+      mountSources: true
+      volumeMounts:
+        - name: springbootpvc
+          path: /data
+  - container:
+      name: runtime
+      image: maysunfaisal/springbootruntime
+      memoryLimit: 768Mi
+      command: ['tail']
+      args: [ '-f', '/dev/null']
+      endpoints:
+        - name: '8080/tcp'
+          exposure: public
+          targetPort: 8080
+          protocol: https
+          secure: true
+      mountSources: false
+      volumeMounts:
+        - name: springbootpvc
+          path: /data
+commands:
+  - exec:
+      id: devBuild
+      commandLine: "/artifacts/bin/build-container-full.sh"
+      component: tools
+      workingDir: /projects/springbootproject
+      group:
+        kind: build
+        isDefault: true
+  - exec:
+      id: devRun
+      commandLine: "/artifacts/bin/start-server.sh"
+      component: runtime
+      workingDir: /
+      group:
+        kind: run
+        isDefault: true
+```
+
+> Download the application
+
+```sh
+$ odo create nodejs --starter
+```
+
+> Create a URL and push the service
+
+```sh
+$ odo url create
+
+$ odo push
+```
+
+**odo**
+
+odo is a fast, iterative, and straightforward CLI tool for developers who write, build, and deploy applications on Kubernetes and OpenShift.
```

Comment:
Should odo be inside backticks here - `odo`? If yes, the change will cascade at other places and other commands (like `oc`) as well.